### PR TITLE
[GSoC] Fixes Sum.doit() with Randomindexed symbol

### LINKED ISSUE #19456: Sum.doit() gives incorrect results with Random Indexed Variables
While working on #19428, I came across the following error
```python
>>> from sympy.stats import *
>>> from sympy import symbols, S, Eq, Sum
>>> B = BernoulliProcess('B', S(1)/2, 1, -1)
>>> t = symbols('t', positive=True, integer=True)
>>> expr = Sum(B[t], (t,0,5))
>>> expr
Sum(B[t], (t, 0, 5))
>>> expr.doit()  # this should not consider B[t] equal in all the cases as it is random variable
6*B[t]          # incorrect result
```
Expected behaviour, it should produce: `B[0] + B[1] + B[2] + B[3] + B[4] + B[5]`

### PR DESCRIPTION
**Pull request by Smit-create**
#### References to other Issues or PRs

Fixes #19456 

#### Brief description of what is fixed or changed
Before fixing:

```python
>>> from sympy.stats import *
>>> from sympy import symbols, S, Eq, Sum
>>> B = BernoulliProcess('B', S(1)/2, 1, -1)
>>> t = symbols('t', positive=True, integer=True)
>>> expr = Sum(B[t], (t, 0, 5))
>>> expr.doit()  # this should not consider B[t] equal in all the cases as it is random variable
6*B[t]       # wrong result
```

After fixing:
```python
>>> from sympy.stats import *
>>> from sympy import symbols, S, Eq, Sum
>>> B = BernoulliProcess('B', S(1)/2, 1, -1)
>>> t = symbols('t', positive=True, integer=True)
>>> expr = Sum(B[t], (t, 0, 5))
>>> expr.doit()
B[0] + B[1] + B[2] + B[3] + B[4] + B[5]
```

#### Release Notes

* stats
   * FIxes free_symbols method of RandomIndexedSymbol

ping @czgdp1807 @Upabjojr @jmig5776

### CODE REVIEW

**Comment by Smit-create** [comment]
@czgdp1807 @Upabjojr Does this look good to go in?

**Comment by czgdp1807** [comment]
LGTM. Will merge by tomorrow, if @Upabjojr @jmig5776 are okay with the patch.

**Comment by jmig5776** [comment]
The patch looks okay to me.

File: sympy/stats/tests/test_stochastic_process.py [hunk 1]
```diff
@@ -194,3 +194,14 @@ def test_BernoulliProcess():
...
+    # test issue 19456
+    expr = Sum(B[t], (t, 0, 4))
+    expr2 = Sum(B[t], (t, 1, 3))
+    expr3 = Sum(B[t]**2, (t, 1, 3))
+    assert expr.doit() == B[0] + B[1] + B[2] + B[3] + B[4]
+    assert expr2.doit() == Y
+    assert expr3.doit() == B[1]**2 + B[2]**2 + B[3]**2
+    assert B[2*t].free_symbols == {B[2*t], t}
+    assert B[4].free_symbols == {B[4]}
+    assert B[x*t].free_symbols == {B[x*t], x, t}
```
Comment (Upabjojr) [inline sympy/stats/tests/test_stochastic_process.py:207]: When doing it on `IndexedBase`, I get:
```python
In [4]: (A[i]).free_symbols
Out[4]: {A, i, A[i]}

In [5]: (A[i]).expr_free_symbols
Out[5]: {A[i]}
```

Maybe this should return `{B[x*t], x, t, B}` ? Did you also test `.expr_free_symbols`? `.expr_free_symbols` is like `.free_symbols` except it stops as soon as the expression tree is not an expression/math formula anymore.

Comment (Smit-create) [reply sympy/stats/tests/test_stochastic_process.py:207]: > Maybe this should return `{B[x*t], x, t, B}`

IMHO, `B` should not be included in free symbols, as it is not an `Expr` object. Also, `expr_free_symbols` works as expected:
```python
>>> B[x*t].expr_free_symbols
{B[t*x]}
>>> B[4].expr_free_symbols
{B[4]}
>>> B[2*t].expr_free_symbols
{B[2*t]}
```

Comment (Smit-create) [reply sympy/stats/tests/test_stochastic_process.py:207]: ping @Upabjojr

Comment (czgdp1807) [reply sympy/stats/tests/test_stochastic_process.py:207]: How would it affect the fix if we return `B` as well?

Comment (Smit-create) [reply sympy/stats/tests/test_stochastic_process.py:207]:
```python
>>> Eq(B[1], 0).as_set()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/smit/Smitlunagariya/sympy/sympy/logic/boolalg.py", line 165, in as_set
    raise NotImplementedError("Sorry, as_set has not yet been"
NotImplementedError: Sorry, as_set has not yet been implemented for multivariate expressions
```

Comment (czgdp1807) [reply sympy/stats/tests/test_stochastic_process.py:207]: Okay, so may be we can exclude `B` from the results.

**Comment by czgdp1807** [comment]
Will be merged tomorrow if no objections raised.

File: sympy/stats/tests/test_stochastic_process.py (hunk 1 above)
Comment (Upabjojr) [reply sympy/stats/tests/test_stochastic_process.py:207]: Maybe `Indexed` should be modified not to return its base element in its free symbols... but that's another issue.

**Pull request merged**